### ISSUE TITLE
dataset type for binary segmentation

### ISSUE THREAD

**Issue by jayayy** [qubvel/segmentation_models#221]
i have images and their background masks in png binary. i need to remove the background and the masks show the object. it essentially is binary segmentation. do i need to add background mask too?

**Comment by jayayy** [comment]
![1-ER1218-1-mask](https://user-images.githubusercontent.com/56384845/66548570-22d02e80-eb5f-11e9-86a6-c4299fcdc79a.png)
![1-ER1218-1](https://user-images.githubusercontent.com/56384845/66548623-3f6c6680-eb5f-11e9-9422-0a01ef1901ce.jpg)

these are the images im hoping the model to train upon. if im making a mistake in generating the data, please guide me to the most efficient way too.

thanks in advance

**Comment by JordanMakesMaps** [comment]
I think in this situation, you'd be doing a binary classification (sigmoid activation function) where your two classes are earring and not-earrings (or 'background', as you said). So the masks you need to create would have 0's on all of the pixels that do not have earrings, and 1's where there are earrings; the masks should be a single channel, not RGB.

Then with those masks prepped, you can use Keras' `to_categorical` function to create one-hot-encoded masks that can be passed to the model. As a sanity check, their shape should be (batch size, height, width, number of classes).

Check out @qubvel example [here](https://github.com/qubvel/segmentation_models/blob/master/examples/binary%20segmentation%20(camvid).ipynb)

**Comment by jayayy** [comment]
just confirming, do i need another mask with background as 1 and object as 0?

**Comment by JordanMakesMaps** [comment]
> just confirming, do i need another mask with background as 1 and object as 0?

Nope, just a single mask for each image, where the object is represented by 1's and background is represented by zeros.

**Comment by jayayy** [comment]
thank you. this has been helpful.
